bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdataframe.py

```diff
@@ -277,19 +282,24 @@ def GetValue(self):
 
         if self._py_arrays is None:
             import numpy
-            from ROOT._pythonization._rdf_utils import ndarray
+            from ROOT._pythonization._rdf_utils import ndarray, all_same_length, is_templated_instance
 
             # Convert the C++ vectors to numpy arrays
             self._py_arrays = {}
             for column in self._columns:
                 cpp_reference = self._result_ptrs[column].GetValue()
                 if hasattr(cpp_reference, "__array_interface__"):
-                    tmp = numpy.asarray(cpp_reference) # This adopts the memory of the C++ object.
+                    tmp = numpy.asarray(cpp_reference)  # This adopts the memory of the C++ object.
                     self._py_arrays[column] = ndarray(tmp, self._result_ptrs[column])
                 else:
                     tmp = numpy.empty(len(cpp_reference), dtype=object)
                     for i, x in enumerate(cpp_reference):
-                        tmp[i] = x # This creates only the wrapping of the objects and does not copy.
+                        if any(is_templated_instance(x, class_name) for class_name in
+                               ["cppyy.gbl.std.vector", "cppyy.gbl.ROOT.VecOps.RVec"]) and all_same_length(x):
```

Comment:
`all_same_lenght` works only if the iterable passed to it is more-than-1-dim IIUC, i.e.

```
auto x = ROOT::RVecI{1,2,3};
auto y = std::vector<int>{1,2,3};
```
These will not work with that function, as `len(next(iter(iterable)))` will return a TypeError because the iterable will give a single integer in this case, of which we can't compute a `len`. Instead this function expects something like
```
auto x = ROOT::RVec<ROOT::RVecI>{{1,2},{3,4},{5,6}};
```
To work properly.